src/backend/utils/misc/guc.c

```diff
@@ -13075,4 +13083,51 @@ guc_set_stack_value(struct config_generic *gconf, config_var_value *val)
 	set_stack_value(gconf,  val);
 }
 
+static char*
+transaction_isolation_to_string(int v)
+{
+	switch (v)
+	{
+		case XACT_SERIALIZABLE:
+			return "serializable";
+		case XACT_REPEATABLE_READ:
+			return "repeatable read";
+		case XACT_READ_COMMITTED:
+			return "read committed";
+		case XACT_READ_UNCOMMITTED:
+			return "read uncommitted";
+	}
+	return "(unrecognized)";
+}
+
+static bool
+check_default_transaction_isolation(int *newval, void **extra, GucSource source)
```

Comment:
any reason for having this as a separate function? can we put it new_val hook?

Reply:
So having a check hook will make sure that we set both 'default_transaction_Isolation' and 'transaction_isolation' together and never just 'default_transaction_Isolation'